WordPress/openverse
# Fix homepage layout shift

## Fixes



Fixes #4815 by @sarayourfriend

## Description


This PR fixes the layout shift for the first-time visitors on the desktop. 
The gallery element is rendered conditionally, only on the client (using ClientOnly). Its width was computed correctly, but since the margins were set on the conditionally on the element itself, its width was shifting. 
The homepage content (both left side with the search bar, and the right side with the gallery) was centered horizontally, so before the gallery was rendered, the left side was centered on the screen. Then, after the gallery rendered, it would shift to the left. I added `justify-between` so the left side is always left-aligned. Since there's no gap between the two sides, the eventual layout with both `justify-between` and `justify-center` is the same, but with the approach in this PR, there is no shift.

Here's the video of the change: first, the current shift in prod, and then - the fix from this PR.

https://github.com/user-attachments/assets/13891b34-55f1-475e-af4e-183bb60a2a20


## Testing Instructions

Run the app, clean the cookies and go to the homepage "http://localhost:8443". Note that the layout shift is minimized.

## Checklist


      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
